### ISSUE TITLE
BrcmPatchRAM crash during startup

### ISSUE THREAD

**Issue by RehabMan** [RehabMan/OS-X-BrcmPatchRAM#5]
Problem is happening in BrcmPatchRAM::publishFirmwareStorePersonality()

```
Thu Jul 30 13:13:31 2015

*** Panic Report ***
panic(cpu 0 caller 0xffffff8000417cc2): Kernel trap at 0xffffff7f829613bf, type 13=general protection, registers:
CR0: 0x000000008001003b, CR2: 0x000000010bb0d000, CR3: 0x000000000408d000, CR4: 0x00000000001626e0
RAX: 0xffffff801e80b2c0, RBX: 0xffffff801e700200, RCX: 0x0000000000000001, RDX: 0x7fffffffffffffff
RSP: 0xffffff81defd3e10, RBP: 0xffffff81defd3e40, RSI: 0x0000000000000000, RDI: 0xffffff801e80b280
R8:  0x0000000000002233, R9:  0x0000000000000002, R10: 0x000000000000005d, R11: 0x0000000000000000
R12: 0xffffff801e93aa40, R13: 0xffffff8000b14568, R14: 0xffffff801e99b0c0, R15: 0xffffff801e80b280
RFL: 0x0000000000010246, RIP: 0xffffff7f829613bf, CS:  0x0000000000000008, SS:  0x0000000000000010
Fault CR2: 0x000000010bb0d000, Error code: 0x0000000000000000, Fault CPU: 0x0

Backtrace (CPU 0), Frame : Return Address
0xffffff81bef3fc70 : 0xffffff800032bda1 
0xffffff81bef3fcf0 : 0xffffff8000417cc2 
0xffffff81bef3feb0 : 0xffffff8000434b73 
0xffffff81bef3fed0 : 0xffffff7f829613bf 
0xffffff81defd3e40 : 0xffffff7f8295ff9e 
0xffffff81defd3e60 : 0xffffff7f8295fe77 
0xffffff81defd3eb0 : 0xffffff80008b38cf 
0xffffff81defd3f30 : 0xffffff80008ae993 
0xffffff81defd3f70 : 0xffffff80008b4883 
0xffffff81defd3fb0 : 0xffffff80004125b7 
      Kernel Extensions in backtrace:
         com.no-one.BrcmPatchRAM(1.7.3)[56552FB6-16A0-353E-B536-093276D588A6]@0xffffff7f8295e000->0xffffff7f82968fff
            dependency: com.apple.iokit.IOUSBFamily(720.4.4)[2A7D7E5E-EEDA-3BF7-B302-0727DCABA6F9]@0xffffff7f8103b000

BSD process name corresponding to current thread: kernel_task
Boot args: kext-dev-mode=1 rootless=0 dart=0 slide=0 

Mac OS version:
14E46

Kernel version:
Darwin Kernel Version 14.4.0: Thu May 28 11:35:04 PDT 2015; root:xnu-2782.30.5~1/RELEASE_X86_64
Kernel UUID: E3C26B2F-8B97-3F1D-B193-690F7E34F830
__HIB  text base: 0xffffff8000100000
System model name: MacBookPro9,2 (Mac-6F01561E16C75D06)

System uptime in nanoseconds: 701057515727
last loaded kext at 701009354118: com.no-one.BrcmPatchRAM   1.7.3 (addr 0xffffff7f8295e000, size 45056)
last unloaded kext at 326449299256: com.apple.filesystems.msdosfs   1.10 (addr 0xffffff7f8295a000, size 61440)
loaded kexts:
com.no-one.BrcmPatchRAM 1.7.3
com.darwin.driver.ACPIBacklight 3.0.0d1
com.insanelymac.RealtekRTL8111  1.2.3
com.jmicron.driver.JMB38X   1.5.0
org.rehabman.voodoo.driver.PS2Trackpad  1.8.15
org.rehabman.voodoo.driver.PS2Keyboard  1.8.15
org.rehabman.driver.FakePCIID   1.2.0
org.hwsensors.driver.ACPISensors    1655
org.rehabman.voodoo.driver.PS2Controller    1.8.15
org.hwsensors.driver.CPUSensors 1655
org.rehabman.driver.AppleSmartBatteryManager    1.55
org.netkas.driver.FakeSMC   1655
com.apple.driver.AppleUSBCDC    4.3.3b1
com.apple.driver.AudioAUUC  1.70
com.apple.driver.AGPM   110.19.6
com.apple.driver.X86PlatformShim    1.0.0
com.apple.filesystems.autofs    3.0
com.apple.driver.AppleUpstreamUserClient    3.6.1
com.apple.driver.AppleHDA   999.69fc1
com.apple.driver.AppleOSXWatchdog   1
com.apple.iokit.IOUserEthernet  1.0.1
com.apple.driver.AppleIntelHD4000Graphics   10.0.6
com.apple.driver.AppleLPC   1.7.3
com.apple.driver.AppleSMCPDRC   1.0.0
com.apple.driver.AppleMCCSControl   1.2.12
com.apple.driver.AppleIntelFramebufferCapri 10.0.6
com.apple.iokit.IOBluetoothSerialManager    4.3.5f8
com.apple.Dont_Steal_Mac_OS_X   7.0.0
com.apple.driver.AppleHWAccess  1
com.apple.driver.AppleHV    1
com.apple.driver.AppleIntelSlowAdaptiveClocking 4.0.0
com.apple.AppleFSCompression.AppleFSCompressionTypeDataless 1.0.0d1
com.apple.AppleFSCompression.AppleFSCompressionTypeZlib 1.0.0
com.apple.BootCache 36
com.apple.iokit.SCSITaskUserClient  3.7.5
com.apple.driver.XsanFilter 404
com.apple.driver.AppleUSBHub    705.4.2
com.apple.iokit.IOAHCIBlockStorage  2.7.1
com.apple.driver.AirPort.Brcm4360   930.37.3
com.apple.driver.AppleUSBEHCI   705.4.14
com.apple.driver.AppleUSBXHCI   710.4.11
com.apple.driver.AppleAHCIPort  3.1.2
com.apple.driver.AppleACPIButtons   3.1
com.apple.driver.AppleHPET  1.8
com.apple.driver.AppleRTC   2.0
com.apple.driver.AppleACPIEC    3.1
com.apple.driver.AppleSMBIOS    2.1
com.apple.driver.AppleAPIC  1.7
com.apple.driver.AppleIntelCPUPowerManagementClient 218.0.0
com.apple.nke.applicationfirewall   161
com.apple.security.quarantine   3
com.apple.security.TMSafetyNet  8
com.apple.driver.AppleIntelCPUPowerManagement   218.0.0
com.apple.kext.triggers 1.0
com.apple.driver.DspFuncLib 272.18.1
com.apple.kext.OSvKernDSPLib    1.15
com.apple.iokit.IOSurface   97.4
com.apple.iokit.IOUSBHIDDriver  705.4.0
com.apple.driver.X86PlatformPlugin  1.0.0
com.apple.driver.IOPlatformPluginFamily 5.9.1d7
com.apple.driver.AppleSMBusController   1.0.13d1
com.apple.driver.AppleHDAController 272.18.1
com.apple.iokit.IOHDAFamily 272.18.1
com.apple.iokit.IOAudioFamily   203.3
com.apple.vecLib.kext   1.2.0
com.apple.iokit.IOUSBUserClient 705.4.0
com.apple.iokit.IONDRVSupport   2.4.1
com.apple.iokit.IOAcceleratorFamily2    156.16
com.apple.AppleGraphicsDeviceControl    3.10.24
com.apple.iokit.IOGraphicsFamily    2.4.1
com.apple.iokit.IOSerialFamily  11
com.apple.iokit.IOBluetoothFamily   4.3.5f8
com.apple.driver.AppleSMC   3.1.9
com.apple.iokit.IOSlowAdaptiveClockingFamily    1.0.0
com.apple.driver.AppleUSBComposite  705.4.9
com.apple.iokit.IOSCSIMultimediaCommandsDevice  3.7.5
com.apple.iokit.IOBDStorageFamily   1.7
com.apple.iokit.IODVDStorageFamily  1.7.1
com.apple.iokit.IOCDStorageFamily   1.7.1
com.apple.iokit.IOAHCISerialATAPI   2.6.1
com.apple.iokit.IOSCSIArchitectureModelFamily   3.7.5
com.apple.iokit.IO80211Family   730.60
com.apple.driver.mDNSOffloadUserClient  1.0.1b8
com.apple.iokit.IONetworkingFamily  3.2
com.apple.iokit.IOUSBFamily 720.4.4
com.apple.iokit.IOAHCIFamily    2.7.5
com.apple.driver.AppleEFINVRAM  2.0
com.apple.driver.AppleEFIRuntime    2.0
com.apple.iokit.IOHIDFamily 2.0.0
com.apple.iokit.IOSMBusFamily   1.1
com.apple.security.sandbox  300.0
com.apple.kext.AppleMatch   1.0.0d1
com.apple.driver.AppleKeyStore  2
com.apple.driver.AppleMobileFileIntegrity   1.0.5
com.apple.driver.AppleCredentialManager 1.0
com.apple.driver.DiskImages 397
com.apple.iokit.IOStorageFamily 2.0
com.apple.iokit.IOReportFamily  31
com.apple.driver.AppleFDEKeyStore   28.30
com.apple.driver.AppleACPIPlatform  3.1
com.apple.iokit.IOPCIFamily 2.9
com.apple.iokit.IOACPIFamily    1.4
com.apple.kec.Libm  1
com.apple.kec.pthread   1
com.apple.kec.corecrypto    1.0
Model: MacBookPro9,2, BootROM MBP91.00D3.B08, 2 processors, Intel Core i5, 2.59 GHz, 16 GB, SMC 2.2f44
Graphics: Intel HD Graphics 4000, Intel HD Graphics 4000, Built-In
Memory Module: BANK0/DIMM0, 8 GB, DDR3, 1600 MHz, Kingston, 9905428-200.A00LF
Memory Module: BANK1/DIMM0, 8 GB, DDR3, 1600 MHz, Kingston, 99U5428-078.A00LF
AirPort: spairport_wireless_card_type_airport_extreme (0x14E4, 0x2123), Broadcom BCM43xx 1.0 (7.15.166.24.3)
Serial ATA Device: TOSHIBA MQ01ABD100, 1 TB
Serial ATA Device: hp      DVD RAM UJ8E0
USB Device: Hub
USB Device: Hub
USB Device: USB Receiver
USB Device: Vendor-Specific Device
USB Device: HP HD Webcam [Fixed]
Thunderbolt Bus: 
```

**Comment by RehabMan** [comment]
This should be fixed in 2015-07-31 build, v1.7.4.

**Comment by RehabMan** [comment]
Confirmed fixed.
